### DISCUSSION TITLE
Any support for offline mobile applications

### DISCUSSION THREAD

**Discussion by jdicami** [openfga/community#266] [Q&A]
Hello,

I've been searching for information or best practices on integrating OpenFGA with offline-first mobile applications. To provide context, our mobile application syncs with a backend database. The web team is currently transitioning our old permission system, based on casl, to FGA. However, I haven't come across any mobile SDKs that enable on-the-fly permission checks on the mobile side. While I recognize that FGA has a more powerful permission model, I'm seeking advice or documentation on how to implement this for our specific use case. Any guidance you can provide would be greatly appreciated.

Thank you.

**Comment by aaguiarz** [comment]
Hi @jdicami 

OpenFGA does not have built-in support for that use case. 

A possible path would be for you to store the results of the authorization checks in a local database, and use those when the application is offline. 

Do you think that could work?
